README.md

```diff
@@ -1 +1,375 @@
+{
+  "name": "Alonzo Church",
+  "birthday": {
+    "/": {
+      "cid": null, // Inherits encoding & hash algorithm from container
+      "dag": {
+        "day": 14,
+        "month": 6
+      }
+    }
+  }
+}
+
+// Or omit the "cid" field
+
+{
+  "name": "Alonzo Church",
+  "birthday": {
+    "/": {
+      "dag": {
+        "day": 14,
+        "month": 6
+      }
+    }
+  }
+}
+```
+
+Inherited encoding is helpful in many situations. A core intention of IPLD is to abstract away from the exact data encoding. When including a link in IPLD, a hash algorithm and encoding MUST be hardcoded. Such choices are often premature, and lead to awkward design choices to avoid this encoding, or exceptions in prose.
+
+Inheriting from the surrounding context enables a level of flexibility for the nested data. Its hash and encoding MUST be the same as the parent, which is by far the most common case. In order to calculate the hash of the entire structure, the direct parent's encoding MUST be known, then passed to the encoder for the nested graph, the CID calculated, and placed into the parent. See more in [IPLD Canonicalization].
+
+The implicit strategy MAY lead to cases where the same DAG generates different CIDs in the same structure, based on different parent encoding contexts.
+
+## 2.3 Capsule
+
+To signal that a payload contains inlined links, the following capsule MAY be used:
+
+``` ipldsch
+type InlineLinkCapsule struct {
+  c Any (rename "ipld/inline/v0.1")
+}
+```
+
+# 3 Encoding Strategies
+
+> In principle, inlining is dead simple: just replace the call of a function by an instance of its body. But any compiler-writer will tell you that inlining is a black art, full of delicate compromises that work together to give good performance without unnecessary code bloat.
+>
+> — [SPJ] & [Simon Marlow], [Secrets of the Glasgow Haskell Compiler inliner][GHC Secrets]
+
+Encoding is trivial on many structures, such as linked lists and trees. The existence of diamond graphs pose a special problem: how many times should a linked graph be inlined versus referenced?
+
+```mermaid
+flowchart
+    a --> b
+    a --> c
+    b -.->|ref?| d
+    c -.->|ref?| d
+    d --> e
+```
+
+There are two basic strategies that take advantage of inlining: redundancy and [spanning tree]s. When inlining is not used, the strategy is a form of tabling ([CAR] files and blockstores), and are included here for completeness.
+
+| Representation | Inlining Strategy   | Space | Traversal                      | Typical Implementation      |
+|----------------|---------------------|-------|--------------------------------|-----------------------------|
+| Redundant Tree | Always              | Large | Fast                           | Standard JSON, CBOR, etc    |
+| Spanning Tree  | Once per unique CID | Small | Often slow; depends on content | DAG-JSON, DAG-CBOR, etc     |
+| Table          | Never               | Small | Medium                         | [CAR] file, blockstore, etc |
```

Comment:
I'd argue "Space" should be "Medium" for the table case. Think of a case with lots of very small blocks (let's say a binary Merkle tree), then you'll end up with roughly double space usage compared to the redundant tree method.

```suggestion
| Table          | Never               | Medium | Medium                         | [CAR] file, blockstore, etc |
```